AISKU/src/ApplicationInsightsDeprecated.ts

```diff
@@ -0,0 +1,317 @@
+        this.appInsightsNew = appInsightsNew;
+    }
+
+    startTrackPage(name?: string) {
+        this.appInsightsNew.startTrackPage(name);
+    }
+
+    stopTrackPage(name?: string, url?: string, properties?: { [name: string]: string; }, measurements?: { [name: string]: number; }) {
+        this.appInsightsNew.stopTrackPage(name, url, properties); // update
+    }
+
+    trackPageView(name?: string, url?: string, properties?: Object, measurements?: Object, duration?: number) {
+        let telemetry: IPageViewTelemetry = {
+            name: name,
+            uri: url
+        };
+
+        // fix for props, measurements, duration
+        this.appInsightsNew.trackPageView(telemetry);
+    }
+
+    trackEvent(name: string, properties?: Object, measurements?: Object) {
+        this.appInsightsNew.trackEvent(<IEventTelemetry>{ name: name});
+    }
+    trackDependency(id: string, method: string, absoluteUrl: string, pathName: string, totalTime: number, success: boolean, resultCode: number) {
+        this.appInsightsNew.trackDependencyData(
+            <IDependencyTelemetry>{ 
+                id: id, 
+                absoluteUrl: absoluteUrl, 
+                commandName: pathName, 
+                duration: totalTime,
+                method: method, 
+                success: success, 
+                resultCode: resultCode
+            });
+    }
+
+    trackException(exception: Error, handledAt?: string, properties?: { [name: string]: string; }, measurements?: { [name: string]: number; }, severityLevel?: any) {
+        this.appInsightsNew.trackException(<IExceptionTelemetry>{
+            error: exception
+        });
+    }
+
+    trackMetric(name: string, average: number, sampleCount?: number, min?: number, max?: number, properties?: { [name: string]: string; }) {
+        this.appInsightsNew.trackMetric(<IMetricTelemetry>{name: name, average: average, sampleCount: sampleCount, min: min, max: max});
+    }
+
+    trackTrace(message: string, properties?: { [name: string]: string; }, severityLevel?: any) {
+        this.appInsightsNew.trackTrace(<ITraceTelemetry>{ message: message, severityLevel: severityLevel });
+    }
+
+    flush(async?: boolean) {
+        this.appInsightsNew.flush(async);
+    }
+
+    setAuthenticatedUserContext(authenticatedUserId: string, accountId?: string, storeInCookie?: boolean) {
+        this.appInsightsNew.setAuthenticatedUserContext(authenticatedUserId, accountId, storeInCookie);
+    }
+
+    clearAuthenticatedUserContext() {
+        this.appInsightsNew.clearAuthenticatedUserContext();
+    }
+    
+    _onerror(message: string, url: string, lineNumber: number, columnNumber: number, error: Error) {
+        this.appInsightsNew._onerror(<IAutoExceptionTelemetry>{ message: message, url: url, lineNumber: lineNumber, columnNumber: columnNumber, error: error });
+    }
+    
+    
+    startTrackEvent(name: string) {
+        throw new Error("Method not implemented.");
```

Comment:
How do we want to handle methods that don't exist in new sdk? If this method is rarely called in someone's code and they dont look for it, it will crash at a potentially bad time. Do we want to remove this so it fails at compile/intellisense time instead? or show a warning?

Reply:
also if these are deprecated, should document as such here (and docs, interface)

Reply:
I added support for startTrackEvent, stopTrackEvent. I think there are valid use case scenario for this api.